src/plugins/discover/public/plugin.tsx

```diff
@@ -180,6 +181,7 @@ export interface DiscoverSetupPlugins {
 export interface DiscoverStartPlugins {
   dataViews: DataViewsServicePublic;
   dataViewEditor: DataViewEditorStart;
+  dataVisualizer?: DataVisualizerPluginStart;
```

Comment:
I'm curious if this needs to be optional? We now have conditional logic around the view toggle, etc. where we didn't before, but do we expect it to ever actually be undefined?

Reply:
I think it's fine as an optional service since: 
1) User can already optionally disabled Field statistics in advanced settings or disable data_visualizer plugin in the yml config 
2) We want it so that the lack of dataVisualizer's service will not break the Discover's basic experience.

Reply:
Oh yeah, disabling the `data_visualizer` plugin would do it. Makes sense, thanks!